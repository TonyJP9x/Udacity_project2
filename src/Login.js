import React, { useEffect, useState } from 'react';
import loginImage from '../src/img/login.jpg';
import * as _DATA from '../src/_DATA'
import { useDispatch, useSelector } from 'react-redux';
import { getUsers, login } from './Slices/AuthSlice';
import { useNavigate } from "react-router-dom";




function Login(props) {
    const [userName, setUserName] = useState('');
    const [password, setPassword] = useState('');
    const  [success, setSuccess] = useState(false)
    const dispatch = useDispatch();
    const navigate = useNavigate();
    const userInfo = useSelector(state => state.login)



    useEffect(() => {
        dispatch(getUsers())
      
    },[])

    const handleOnSubmit = (e) =>{
        e.preventDefault();
        const userInfoValue = Object.values(userInfo.value)
        for (let i = 0; i <userInfoValue.length; i++ ){
            if(userName === userInfoValue[i]['id']){
                if(password === userInfoValue[i]['password']){
                    setSuccess(true)
                    dispatch(login(userInfoValue[i]))
                 setTimeout(() =>{
                     navigate('/home')
                 }, 1000)
                    
                }
            }
     
        }    
    }
    return (
        <div>
            <img  src={loginImage}/>
     
           <h1 data-testid='message' style={{color: 'green'}}>{success? 'Login successfully !' : null}</h1>
     
            <form >
                <h1>Login </h1>
                <h4>User</h4>
                <input data-testid='username' type='text' placeholder='User' style={{'width':'500px'}}  required  value={userName} onChange={(e) => setUserName(e.target.value)} />
                <h4>Password</h4>
                <input data-testid='password' type='text' placeholder='Password' style={{'width':'500px', marginBottom:'20px'}} required  value={password} onChange={(e) => setPassword(e.target.value)}/><br/>
                <button data-testid='button' className='btn btn-primary' type='button' onClick={(e) => handleOnSubmit(e)}>Submit </button>
            </form >
        </div>
    );
}

export default Login;

