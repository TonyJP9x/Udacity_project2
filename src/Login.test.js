import { fireEvent, render } from "@testing-library/react";
import * as React from "react";
import Login from "./Login";
import { Provider } from "react-redux";
import store from "./Store";
import { MemoryRouter } from "react-router-dom";
import { waitFor } from "@testing-library/react";




describe("Login", () => {
  it("will have all expected field", () => {
    var component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it('will show message if all mandatory field has been filled', async () => {
    var component = render(
        <MemoryRouter>
          <Provider store={store}>
            <Login />
          </Provider>
        </MemoryRouter>
      );
    var option1 = component.getByTestId('username')
    fireEvent.change(option1, {target: {value:'sarahedo'}})
    var option2 = component.getByTestId('password')
    fireEvent.change(option2, {target: {value:'password123'}})
    var button = component.getByTestId('button')
    fireEvent.click(button)
    await waitFor(() => {
        expect(component.getByTestId('message')).toBeInTheDocument();
    })


})
});
