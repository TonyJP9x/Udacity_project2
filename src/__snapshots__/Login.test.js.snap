// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login will have all expected field 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <img
          src="login.jpg"
        />
        <h1
          data-testid="message"
          style="color: green;"
        />
        <form>
          <h1>
            Login 
          </h1>
          <h4>
            User
          </h4>
          <input
            data-testid="username"
            placeholder="User"
            required=""
            style="width: 500px;"
            type="text"
            value=""
          />
          <h4>
            Password
          </h4>
          <input
            data-testid="password"
            placeholder="Password"
            required=""
            style="width: 500px; margin-bottom: 20px;"
            type="text"
            value=""
          />
          <br />
          <button
            class="btn btn-primary"
            data-testid="button"
            type="button"
          >
            Submit 
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <img
        src="login.jpg"
      />
      <h1
        data-testid="message"
        style="color: green;"
      />
      <form>
        <h1>
          Login 
        </h1>
        <h4>
          User
        </h4>
        <input
          data-testid="username"
          placeholder="User"
          required=""
          style="width: 500px;"
          type="text"
          value=""
        />
        <h4>
          Password
        </h4>
        <input
          data-testid="password"
          placeholder="Password"
          required=""
          style="width: 500px; margin-bottom: 20px;"
          type="text"
          value=""
        />
        <br />
        <button
          class="btn btn-primary"
          data-testid="button"
          type="button"
        >
          Submit 
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
